### PR TITLE
Story/10733 validation decorator unit tests

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Added unit tests for the Validation Decorator service

## Link to Jira
(https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10733)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/SIDValidationDecoratorResource.swift [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import Foundation
 
-struct ServiceIdentityDocumentValidationDecoratorResource: Resource {
+struct SIDValidationDecoratorResource: Resource {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/SIDValidationDecoratorResource.swift:7]: I would prefer not to introduce abbreviations, I think the name was more precise and easier to understand before

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/SIDValidationDecoratorResourceTests.swift [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		serviceProvider.load(resource) { result in
+			switch result {
+			case .success:
+				XCTFail("client error is expected!")
+			case .failure(let error):
+				guard case let .receivedResourceError(resourceError) = error,
+					  .VD_ID_CLIENT_ERR == resourceError else {
+						  XCTFail("unexpected error case")
+						  return
+					  }
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/SIDValidationDecoratorResourceTests.swift:33]:
```suggestion
				guard case let .receivedResourceError(. VD_ID_CLIENT_ERR) = error else {
						  XCTFail("Expected .VD_ID_CLIENT_ERR but received: \(error)")
						  return
					  }
```

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/SIDValidationDecoratorResourceTests.swift:33]: If you don't compare the error but just capture it (and for alle the remaining tests, too), you will not need to implement the Equatable protocol for `ServiceIdentityValidationDecoratorError`.

**Review by pascal-brause — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/SIDValidationDecoratorResource.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/SIDValidationDecoratorResource.swift:7]: yes i just reverted it now

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/SIDValidationDecoratorResourceTests.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/__tests__/SIDValidationDecoratorResourceTests.swift:33]: yeah you are right, i had it first as i used to compare but now it is obsolete, i will remove the equatable conformance

**Review by pascal-brause — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**